examples/plot_2D_example.py

```diff
@@ -0,0 +1,210 @@
+    roi_size = 4
+    sigma = 2.0
+    smooth_X = 1.0
+
+    # hyper-parameters
+    n_clusters = 200
+
+    # inference parameters
+    fwer_target = 0.1
+    delta = 6
+    # delta = 6 is the tolerance parameter corresponding to
+    # n_clusters = 200 for this particular scenario.
+    # To compute it, one must compute the largest cluster diameter.
+
+    # computation parameter
+    n_jobs = 1
+
+    # computing the thresholds for feature selection
+    correction_no_cluster = 1. / n_features
+    correction_cluster = 1. / n_clusters
+    thr_c = zscore_from_sf((fwer_target / 2) * correction_cluster)
+    thr_nc = zscore_from_sf((fwer_target / 2) * correction_no_cluster)
+
+    X_init, y, beta, epsilon = \
+        simulation_2D(n_samples, shape, roi_size, sigma, smooth_X, seed)
+
+    empirical_snr(X_init, y, beta, epsilon)
+
+    beta_extended = weight_map_2D_extended(shape, roi_size, delta)
+
+    # desparsified lasso
+    beta_hat, cb_min, cb_max = \
+        desparsified_lasso_confint(X_init, y, n_jobs=n_jobs)
+    sf, sf_corr = sf_from_cb(cb_min, cb_max)
+    cdf, cdf_corr = cdf_from_cb(cb_min, cb_max)
+    zscore = zscore_from_sf_and_cdf(sf, cdf)
+    selected_dl = zscore > thr_nc
+    selected_dl = np.logical_or(sf_corr < fwer_target / 2,
+                                cdf_corr < fwer_target / 2)
+
+    # clustered desparsified lasso (CluDL)
+    connectivity = image.grid_to_graph(n_x=shape[0],
+                                       n_y=shape[1])
+    ward = FeatureAgglomeration(n_clusters=n_clusters,
+                                connectivity=connectivity,
+                                linkage='ward')
+    sf, sf_corr, cdf, cdf_corr = \
+        clustered_inference(X_init, y, ward, n_clusters)
+    zscore = zscore_from_sf_and_cdf(sf, cdf)
+    selected_cdl = zscore > thr_c
+    selected_cdl = np.logical_or(sf_corr < fwer_target / 2,
+                                 cdf_corr < fwer_target / 2)
+
+    # ensemble of clustered desparsified lasso (EnCluDL)
+    sf, sf_corr, cdf, cdf_corr = \
+        ensemble_clustered_inference(X_init, y, ward,
+                                     n_clusters, train_size=0.1)
+    zscore = zscore_from_sf_and_cdf(sf, cdf)
+    selected_ecdl = zscore > thr_c
+    selected_ecdl = np.logical_or(sf_corr < fwer_target / 2,
+                                  cdf_corr < fwer_target / 2)
+
+    maps = []
+    titles = []
+
+    maps.append(np.reshape(beta, shape))
+    titles.append('True weights')
+
+    maps.append(np.reshape(beta_extended, shape))
+    titles.append('True weights \nwith tolerance')
+
+    maps.append(np.reshape(selected_dl, shape))
+    titles.append('Desparsified Lasso')
+
+    maps.append(None)
+    titles.append(None)
+
+    maps.append(np.reshape(selected_cdl, shape))
+    titles.append('CluDL')
+
+    maps.append(np.reshape(selected_ecdl, shape))
+    titles.append('EnCluDL')
+
+    plot(maps, titles)
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
no need to protect with main() for sphinx gallery

Reply:
The example is here to explain stuff. So you should provide more comments to guide the user toward the important observations.